pkg/plugins/mellanox/mellanox_plugin.go

```diff
@@ -180,7 +180,13 @@ func (p *MellanoxPlugin) Apply() error {
 		return nil
 	}
 	glog.Info("mellanox-plugin Apply()")
-	return configFW()
+	if err := configFW(); err != nil {
+		return err
+	}
+	if err := resetFW(); err != nil {
```

Comment:
im not too keen on always running mstfwreset, perhaps only for DPUs in Embedded mode ?

i need to think about it a bit more

Reply:
also for DPU you need to run with --reset-sync=1 flag.

see:
https://docs.nvidia.com/networking/pages/viewpage.action?pageId=52009175

with current implementation, does it work for u ?

Reply:
Aahh good to know. I was testing the current implementation on a cluster w/ the BF in NIC mode, which was working. The issue I was seeing was not specific to running in DPU embedded mode, the node would get stuck in a bootloop if I applied a policy to change the number of vfs, since after rebooting the updated configuration would not be applied, triggering another reboot.

Reply:
So, you encountered the issue with DPU in NIC mode as well ?

Reply:
Yes

Reply:
My understanding since submitting this patch is that this is not intended behavior by Nvidia and a firmware patch will come in July to address this issue.

Reply:
The issue is that the behavior of NIC mode changed too and is broken in a similar to DPU. With DPU mode --sync 1 causes the DPU to hang. I have code that adds `--sync 1` but I want it to work correctly before pushing the changes.

Reply:
> My understanding since submitting this patch is that this is not intended behavior by Nvidia and a firmware patch will come in July to address this issue.

that is correct, its a bug and Nvidia should fix the firmware.

@SalDaniele in that case, i think this PR is no longer required and can be closed ?

Reply:
Yes provided this fix is in the next firmware release, we can close this PR